src/test/java/roomescape/reservation/application/service/ReservationCommandServiceTest.java

```diff
@@ -250,4 +200,152 @@ void deleteNonExistentReservation() {
                 .isInstanceOf(NotFoundException.class)
                 .hasMessage("[RESERVATION] not found. params={ReservationId=ReservationId(-1)}");
     }
+
+    @Test
+    @DisplayName("슬롯의 첫 번쨰 예약은 곧바로 승인된다.")
+    void firstReservationOfSlotIsApprove() {
+        // given
+        final Theme theme = themeRepository.save(
+                Theme.withoutId(
+                        ThemeName.from("공포"),
+                        ThemeDescription.from("지구별 방탈출 최고"),
+                        ThemeThumbnail.from("www.making.com")));
+
+        final User user = userRepository.save(
+                User.withoutId(
+                        UserName.from("강산"),
+                        Email.from("email@email.com"),
+                        Password.fromEncoded("1234"),
+                        UserRole.NORMAL));
+
+        final User somebody = userRepository.save(
+                User.withoutId(
+                        UserName.from("다른사람"),
+                        Email.from("email@email.com"),
+                        Password.fromEncoded("1234"),
+                        UserRole.NORMAL));
+
+        final ReservationDate date = ReservationDate.from(LocalDate.of(2025, 8, 5));
+        final ReservationTime time = ReservationTime.from(LocalTime.of(10, 0));
+
+        final Reservation reservation1 = Reservation.withoutId(
+                user.getId(),
+                date,
+                time,
+                theme
+        );
+
+        final Reservation reservation2 = Reservation.withoutId(
+                somebody.getId(),
+                date,
+                time,
+                theme
+        );
+
+        reservationCommandService.create(reservation1);
+        reservationCommandService.create(reservation2);
+
+        // when
+        // then
+        assertThat(reservation1.getStatus() == BookedStatus.APPROVED).isTrue();
+        assertThat(reservation2.getStatus() == BookedStatus.WAITING).isTrue();
+    }
+
+    @Test
+    @DisplayName("동일 슬롯에 동일 유저로 중복 예약 시 DuplicateException이 발생한다")
+    void cannotReserveSameSlotTwiceBySameUser() {
+        // given
+        final Theme theme = themeRepository.save(Theme.withoutId(
+                ThemeName.from("공포"),
+                ThemeDescription.from("지구별 방탈출 최고"),
+                ThemeThumbnail.from("www.making.com")));
+        final User user = userRepository.save(User.withoutId(
+                UserName.from("강산"),
+                Email.from("email@email.com"),
+                Password.fromEncoded("1234"),
+                UserRole.NORMAL));
```

Comment:
```suggestion
        final Theme theme = saveTheme("공포", "지구별 방탈출 최고", "www.making.com");
        final User user = saveUser("강산", "email@email.com, "www.1234.com", UserRole.NORMAL);
```

메서드를 만들어서 테스트코드의 가독성을 향상시키면 좋아요~

Reply:
감사합니다! 수정하겠습니다.

---

픽스처 클래스를 따로 만들지 않고 그냥 메서드로만 추출해서 써도 괜찮은 건지 고민이 됩니다. 만약 클래스가 점점 늘어나게 되면, 오히려 테스트 복잡도가 더 올라가는 건 아닌가 싶기도 하고요.

이미 작성된 테스트 코드가 많아질 때쯤 되니까, 그때 가서야 테스트 리팩터링이 필요하다는 걸 느끼는데, 사실 이미 좀 늦은 게 아닌가 싶은 생각도 듭니다. 물론 지금은 코드가 얼마 되지 않으니까 전부 변경해야 한다고 해도 부담이 크진 않은데, 테스트가 더 많아진 상태라면 도중에 갑자기 픽스처 클래스를 도입하거나 메서드 추출 방식을 바꾸는 게 과연 가능할까? 이전 테스트 코드를 그대로 두고 새로운 코드부터 작성한다면 일관성에 어긋나는 건 아닌가? 싶기도 해요.

어떻게 보면 테스트 코드가 좀 지저분해지는 건 필연적인 미래이기에, 처음부터 기준을 딱 정해서 가야 하는 건가? 근데 또 애초에 각 잡고 시작하면 오히려 거부감이 생길 수도 있을 것 같고… 이런저런 고민이 계속 생기는 것 같습니다.

Reply:
처음부터 제대로 구조를 잡고 완벽하게 코드를 작성하면 좋겠지만, 세상사 그럴 수가 없죠.
때문에 리팩터링하는 것은 자연스러운 흐름이에요.

프로덕션 코드에 기능을 추가하다가도 어느 순간은 날 잡고 리팩터링하기만 하는데요.
테스트 코드도 마찬가지예요. 필요한 시점이 오면 리팩터링하면 돼요.

Reply:
필요한 시점에 날 잡고 리팩터링!! 경험을 알려주셔서 감사합니다 😄